Refactor: Unify simple and advanced modes (Steps 1B)

maudnals opened the pull request:
* Bugfix on simple mode 
* Use the new simulation object structure in the simple mode
* TBD Use the new simulation object structure in the advanced mode

This PR applies the steps 1B outlined in our [migration plan](https://docs.google.com/document/d/1j9lU5spUYJgE1Qh_vrNIoQ-5l19_gEfEs_-YUmkKYnk).

maudnals commented on public/js/dom.js:
```diff
@@ -1413,6 +1413,149 @@ export function loadPython() {
+    })
+
+    // Save table temporarily; used for XLSX multi-table download
+    tempSaveTable_simpleMode(table, `${simulationId}-${measurementGoal}`)
```
Doesn't this break the download function for the advanced mode?
If so: let's consider replacing the current implementation of the temp tables with a cross-modes unified object for all tables. That object should be emptied (reset to `{}`) upon mode switch.
WDYT?

cucuandreea replied:
This seems ok for the advanced mode. 
However it seems the downloadAll is broken for advanced only

cucuandreea replied:
Ok, fixed this with the latest commit - we have different "tempSaveTable_<mode>" because the table id s have different structure. We could just use the same structure for that as well.
For now I fixed by checking current mode and using the corresponding structure.

cucuandreea commented on public/js/dom.js:
```diff
@@ -1413,6 +1413,149 @@ export function loadPython() {
+
+
+
+export function displaySimulationResults_unified(simulation, mode) {
```
SGTM. However mode is also needed for the table ids which are different, so we will have to adjust that too

maudnals commented:
New set of changes:
* Bugfix: Fix the download bug in advanced mode
* Bugfix: Fix the last parameter issue in `simulate` function
* Refactor: Unified **Save** and **Download** functions across modes
* Refactor: Removed need for `mode` in `displaySimulationResults` + Used one unique `all-simulations-wrapper`
* Refactor: Moved code out of `main`, Created `store` and `simulate` JS files to contain associated functions
* Refactor: Miscellaneous renamings (removed `unified`)
* Refactor: Removed all unused code e.g. mode-specific functions that weren't called anymore
